runtime/j9vm/javanextvmi.cpp

```diff
@@ -681,11 +685,159 @@ JVM_TakeVirtualThreadListToUnblock(JNIEnv *env, jclass ignored)
+		return currentThread;
+	}
+
+	if (isVirtualThread) {
+		vm->internalVMFunctions->acquireVThreadInspector(currentThread, thread, TRUE);
+		/* Re-fetch threadObject since acquireVThreadInspector can release and reacquire VM access. */
+		threadObject = J9_JNI_UNWRAP_REFERENCE(thread);
+	}
+	/* Make sure the vmThread stays alive while it is being used. */
+	f_monitorEnter(vm->vmThreadListMutex);
+	if (isVirtualThread) {
+		jint vthreadState = 0;
+		j9object_t carrierThread = NULL;
+		vthreadState = J9VMJAVALANGVIRTUALTHREAD_STATE(currentThread, threadObject);
+		carrierThread = (j9object_t)J9VMJAVALANGVIRTUALTHREAD_CARRIERTHREAD(currentThread, threadObject);
+		if (NULL != carrierThread) {
+			targetThread = J9VMJAVALANGTHREAD_THREADREF(currentThread, carrierThread);
+		}
+		if (J9OBJECT_I64_LOAD(currentThread, threadObject, vm->virtualThreadInspectorCountOffset) < -2) {
+			/* If the virtual thread is suspended in transition, check if the mounting process is already completed. */
+			J9VMThread *carrierVMThread = VM_VMHelpers::getCarrierVMThread(currentThread, threadObject);
+			if (NULL != carrierVMThread->currentContinuation) {
+				targetThread = carrierVMThread;
+			}
+		}
+		j9object_t continuationObj = J9VMJAVALANGVIRTUALTHREAD_CONT(currentThread, threadObject);
+		ContinuationState continuationState = *VM_ContinuationHelpers::getContinuationStateAddress(currentThread, continuationObj);
+		if ((JVMTI_VTHREAD_STATE_NEW != vthreadState)
+		&& (JVMTI_VTHREAD_STATE_TERMINATED != vthreadState)
+		&& !VM_ContinuationHelpers::isFinished(continuationState)
+		) {
+			*isThreadAlive = JNI_TRUE;
+		}
+	} else {
+		targetThread = J9VMJAVALANGTHREAD_THREADREF(currentThread, threadObject);
+		*isThreadAlive = J9VMJAVALANGTHREAD_STARTED(currentThread, threadObject) && (NULL != targetThread);
+	}
+
+	if (!isThreadAlive) {
+		f_monitorExit(vm->vmThreadListMutex);
+		if (isVirtualThread) {
+			vm->internalVMFunctions->releaseVThreadInspector(currentThread, thread);
+		}
+		return NULL;
+	}
+	if (NULL != targetThread) {
+		targetThread->inspectorCount += 1;
+	}
+
+	f_monitorExit(vm->vmThreadListMutex);
+	return targetThread;
+}
+
+static void
+releaseTargetVMThread(J9VMThread *currentThread, J9VMThread *targetThread, jthread thread, jboolean isVirtualThread)
+{
+	j9object_t threadObject = J9_JNI_UNWRAP_REFERENCE(thread);
+	if ((currentThread->threadObject != threadObject) && isVirtualThread) {
+		currentThread->javaVM->internalVMFunctions->releaseVThreadInspector(currentThread, thread);
+	}
+	if (currentThread != targetThread) {
+		J9JavaVM *vm = targetThread->javaVM;
+		/* Release the J9VMThread (allow it to die) now that we are no longer inspecting it. */
+		f_monitorEnter(vm->vmThreadListMutex);
+		if (0 == --(targetThread->inspectorCount)) {
+			f_monitorNotifyAll(vm->vmThreadListMutex);
+		}
+		f_monitorExit(vm->vmThreadListMutex);
+	}
+}
+
 JNIEXPORT jobject JNICALL
 JVM_CreateThreadSnapshot(JNIEnv *env, jobject thread)
```

Comment:
General statement: these thread natives should really be defined in runtime/jcl/thread.cpp but we dont need to do that in this PR.